software/webapp/src/main/java/org/apache/brooklyn/entity/proxy/AbstractControllerImpl.java

```diff
@@ -317,6 +331,8 @@ protected void computePortsAndUrls() {
 
         sensors().set(PROTOCOL, inferProtocol());
         sensors().set(MAIN_URI, URI.create(inferUrl()));
+        sensors().set(MAIN_URI_MAPPED_SUBNET, URI.create("http://" + sensors().get(SUBNET_ADDRESS) + ":" + sensors().get(PROXY_HTTP_PORT) + "/"));
```

Comment:
Yes, except it uses the slightly weird `Machines. findSubnetHostname`. On balance, we should probably just live with that, so revert it to calling inferUrl!

To get the unit test to work, we'd need to switch to a location type that implemented `HasSubnetHostname` (annoying - find to leave as a todo, to improve the test in the future).